lib/index.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brLedger = require('bedrock-ledger');
+const equihashSigs = require('equihash-signature');
+const validate = require('bedrock-validation').validate;
+const BedrockError = bedrock.util.BedrockError;
+
+require('./config');
+
+bedrock.events.on('bedrock.start', () => brLedger.use('EquihashValidator2017', {
+  type: 'validator',
+  api: api
+}));
+
+const api = {};
+// NOTE: only exported for tests
+module.exports = api;
+
+api.mustValidateEvent = (event, validatorConfig, options, callback) => {
+  if(typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  api.validateConfiguration(validatorConfig, err => {
+    if(err) {
+      return callback(err);
+    }
+    if(!validatorConfig.eventFilter.some(f =>
```

Comment:
Yes, I believe that's true.  This is a good time to mention that now that we have the `requireEventValidation` flag: https://github.com/digitalbazaar/bedrock-ledger/blob/cm3/test/mocha/mock.data.js#L87

A validator like the `EventTypeValidator2017` becomes almost irrelevant to my way of thinking.  When the `requireEventValidation` flag is set to true, if there is no validator configured for a particular event type, the event will be rejected.

There however may be use cases for ONLY validating the event type, in which case `EventTypeValidator2017` would be a useful tool.